### ISSUE TITLE
Error during the Training Step of ResNet 50  -> tensorflow warning: can save best model only with val_acc available, skipping 

### ISSUE THREAD

**Issue by Raghav2305** [zekaouinoureddine/COVID-19-DETECTION-FROM-X-RAY#2]
During the training step, I provided **metric = "val_acc"** and after completion of the first epoch, it gave the error: "tensorflow warning: can save the best model only with val_loss available, skipping". I tried changing the metric to **"val_accuracy"** as defined in the tensorflow documentation but still had the same warning. Because of this the model is training completely but no **best_model.h5** file is being stored. 

![error](https://user-images.githubusercontent.com/82498222/232185403-a58d0e07-4beb-4009-ae48-e1c24cab6ee0.jpg)

**Comment by zekaouinoureddine** [comment]
Hey, In Keras, `val_acc` and `val_accuracy` are equivalent metrics. To fix your issue, I recommend replacing the deprecated function `fit_generator()` with `fit()`, and omitting the arguments **`steps_per_epoch`** and **`validation_steps`**. These changes should help eliminate the warning message and restore the **`best_model.h5`**.

**Before**
```python
history=model.fit_generator(train_datagen.flow(X_train, y_train, batch_size = bs),
                            steps_per_epoch = len(X_train)//bs,
                            validation_data = (X_valid, y_valid),
                            validation_steps = len(X_valid)//bs,
                            epochs =epochs,
                            callbacks= [checkpointer])
```
**After**
```python
history=model.fit(train_datagen.flow(X_train, y_train, batch_size = bs),
                            # steps_per_epoch = len(X_train)//bs,
                            validation_data = (X_valid, y_valid),
                            # validation_steps = len(X_valid)//bs,
                            epochs =epochs,
                            callbacks= [checkpointer])
````

**Comment by Raghav2305** [comment]
Hey, I tried this and it worked perfectly. By solving this issue #3 is also resolved. Thanks for the help. Closing both issues now. Thanks again!
